wallet/src/recovering/paperwallet.rs

```diff
@@ -47,6 +47,70 @@ pub fn unscramble(password: &[u8], input: &[u8]) -> Vec<u8> {
     out
 }
 
+const PAPERWALLET_CERTIFICATE_MNEMONIC_LENGTH: usize = 27;
+const SCRAMBLED_INPUT_WORD_COUNT: usize = 18;
+
+struct InputPassphrase<'a> {
+    input: &'a str,
+    passphrase: &'a str,
+}
+
+// helper function for get_scrambled_input
+fn split_scrambled_input_passphrase(mnemonics: &str) -> Option<InputPassphrase> {
+    use bip39::dictionary::Language;
+    let dic = bip39::dictionary::ENGLISH;
+
+    if mnemonics.split(dic.separator()).count() == PAPERWALLET_CERTIFICATE_MNEMONIC_LENGTH {
+        let scrambled_input_end_position = mnemonics
+            .match_indices(dic.separator())
+            .nth(SCRAMBLED_INPUT_WORD_COUNT - 1)
+            .map(|(idx, _sep)| idx)
+            .unwrap();
+
+        Some(InputPassphrase {
+            input: &mnemonics[..scrambled_input_end_position],
+            passphrase: &mnemonics[scrambled_input_end_position + 1..],
+        })
+    } else {
+        None
+    }
+}
+
+/// daedalus paperwallet is expected to be split from
+/// 27 mnemonic words:
+///
+/// * 18 words of "input"
+/// * 9 words of "password"
+///
+pub fn daedalus_paperwallet(mnemonics: &str) -> Result<Option<bip39::Entropy>, bip39::Error> {
+    let InputPassphrase { input, passphrase } = match split_scrambled_input_passphrase(mnemonics) {
+        Some(input_passphrase) => input_passphrase,
+        None => return Ok(None),
+    };
+    let dic = bip39::dictionary::ENGLISH;
+    let mnemonics = bip39::Mnemonics::from_string(&dic, input)?;
+    let entropy1 = bip39::Entropy::from_mnemonics(&mnemonics)?;
+
+    let mnemonics = bip39::Mnemonics::from_string(&dic, passphrase)?;
+    let entropy2 = bip39::Entropy::from_mnemonics(&mnemonics)?;
+
+    let entropy = unscramble(&entropy2, &entropy1);
+    let entropy = bip39::Entropy::from_slice(&entropy)?;
+
+    /*
+    // TODO: Not sure if the recovered root key is actually an address or a wallet
+    let key = super::from_daedalus_entropy(
+        entropy,
+        ed25519_bip32::DerivationScheme::V1,
+    ).expect("encoding in memory to work")
+    // XXX: .coerce_unchecked::<Rindex<Address>>()
+    ;
+    */
```

Comment:
So this is some question that are still pending here how is it derived from the given entropy

1. add a paperwallet recovering type (just as in daedalus and yoroi);
2. implement the `check_address` function (so we can make a test will help reverse engineer whatever they did here 🤦 )
3. implement the `check_block` function as in the others;
4. implement the dump_in part
5. add function in `RecoveryBuilder` to `build_paperwallet` recovering type newly defined
6. make the wallet-c and wallet-js call `build_paperwallet` and recover from that too